src/filter.jl

```diff
@@ -0,0 +1,148 @@
+- `db`: A RecordDatabase object.
+- `level`: The character string of the level at which to filter the database. Possible values are `"Record"`, `"Study"`, `"Model"`, `"Taxon"`.
+
+# Examples
+```jldoctest filter-examples
+
+## Example just for demonstration, coding is not actually derived from the paper!
+
+using Taxonomy
+using Taxonomy.Judgements
+
+test_db = RecordDatabase()
+
+test_db += Record(
+    rater="NH",
+    id="2a129694-550c-4396-be6f-00507b1dc7ba",
+    location=DOI("10.1007/s10869-019-09648-5"),
+    Lang("en"),
+    Study(
+        N(100, 0.8),
+        Model(Standardized(false),
+            LatentPathmodel(
+                Structural(structural_model=missing),
+                Dict(
+                    :IP => Measurement(n_variables=3, factor_variance=missing, loadings=missing, quest_scale=7),
+                    :IN => Measurement(n_variables=4, factor_variance=missing, loadings=missing, quest_scale=6),
+                    :DN => Measurement(n_variables=3, factor_variance=missing, loadings=missing, quest_scale=7),
+                    :BC => Measurement(n_variables=3, factor_variance=missing, loadings=missing, quest_scale=5),
+                    :IB => Measurement(n_variables=5, factor_variance=missing, loadings=missing, quest_scale=5)
+                )
+            )
+        ),
+        Model(Standardized(true),
+            NoTaxon()
+        )
+    ),
+    Study(
+        N(200, 0.98),
+        Model(Standardized(false),
+        ),
+        Model(Standardized(false))
+    )
+)
+
+## Filtering on Taxon level
+filter_Pathmodel = filter(x -> typeof(x) == NoTaxonEver, test_db, "Taxon")
+Model(Study(filter_Pathmodel[Base.UUID("2a129694-550c-4396-be6f-00507b1dc7ba")])[1])[2]
+
+# output
+Model(Dict{Symbol, Vector{Union{Taxon, AbstractJudgement}}}(:Taxon => [NoTaxonEver
+], :Standardized => [Standardized{Bool}(true, 1.0, missing)]))
+```
+
+```jldoctest filter-examples
+
+## Filtering on Model level
+filter_Standardized = filter(x -> rating(x, :Standardized) == true, test_db, "Model")
+Model(Study(filter_Standardized[Base.UUID("2a129694-550c-4396-be6f-00507b1dc7ba")])[1])
+
+# output
+1-element Vector{Union{Model, AbstractJudgement}}:
+ Model(Dict{Symbol, Vector{Union{Taxon, AbstractJudgement}}}(:Taxon => [NoTaxonEver
+], :Standardized => [Standardized{Bool}(true, 1.0, missing)]))
+
+```
+
+```jldoctest filter-examples
+
+## Filtering on Study level
+filter_N = filter(x -> certainty(x, :N) > 0.9, test_db, "Study")
+Study(filter_N[Base.UUID("2a129694-550c-4396-be6f-00507b1dc7ba")])
+
+
+# output
+1-element Vector{Union{Study, AbstractJudgement}}:
+ Study(Dict{Symbol, Vector{Union{JudgementLevel, AbstractJudgement}}}(:N => [N{Int64}(200, 0.98, missing)]))
+```
+"""
+function Base.filter(f, db::RecordDatabase, level::AbstractString)::RecordDatabase
+
+    @assert level in ["Record", "Study", "Model", "Taxon"] "Invalid level. Choose from 'Record', 'Study', 'Model', 'Taxon'"
```

Comment:
Not quite happy with asserting the filter level like that, do you have a more elegant way?